### PR TITLE
prepare for v0.104.0

### PR DESCRIPTION
**Pull request by jinja2**
**Description:** <Describe what has changed.>

**Link to Splunk idea:** <Link to Splunk idea, see https://ideas.splunk.com>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,20 +2,96 @@
 
 ## Unreleased
 
+## v0.104.0
+
+This Splunk OpenTelemetry Collector release includes changes from the [opentelemetry-collector v0.104.0](https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.104.0) and the [opentelemetry-collector-contrib v0.104.0](https://github.com/open-telemetry/opentelemetry-collector-contrib/releases/tag/v0.104.0) releases where appropriate.
+
+> 🚩 When setting properties for discovery receiver as environment variables (`SPLUNK_DISCOVERY_*`), the values cannot reference other environment variables without curly-braces. For example, user is trying to set discovery property `SPLUNK_DISCOVERY_EXTENSIONS_k8s_observer_ENABLED` to the value of another env var, `K8S_ENVIRONMENT`.
```
Comment (aunshc) [inline CHANGELOG.md:9]: Minor suggestion: I would switch the sequence of the warning and example of the discovery receiver here.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,20 +2,96 @@
...
+> For versions older than 0.104.0, setting it as `SPLUNK_DISCOVERY_EXTENSIONS_k8s_observer_ENABLED=\$K8S_ENVIRONMENT` (note the escaped variable name does not have curly braces) was valid. But from v0.104.0, env var names need to be passed with braces. For this example, user should modify it to `SPLUNK_DISCOVERY_EXTENSIONS_k8s_observer_ENABLED=\${K8S_ENVIRONMENT}`.
+
+> :warning: In our efforts to align with [upstream's goals](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/rfcs/env-vars.md) for environment variable resolution in collector's configuration, the Splunk OpenTelemetry Collector will be dropping support for expansion of BASH-style environment variables, such as `$FOO` in the configuration in an upcoming version. Users are adviced to update collector configuration to use `${env:FOO}` instead.
```
Comment (aunshc) [inline CHANGELOG.md:12]:
```suggestion
> :warning: In our efforts to align with the [goals](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/rfcs/env-vars.md) defined upstream for environment variable resolution in the Collector's configuration, the Splunk OpenTelemetry Collector will be dropping support for expansion of BASH-style environment variables, such as `$FOO` in the configuration in an upcoming version. Users are advised to update their Collector's configuration to use `${env:FOO}` instead.
```

File: CHANGELOG.md (hunk 1 above)
Comment (jinja2) [reply CHANGELOG.md:11]: done

**Review by atoulme — APPROVED** [review]

**Review by atoulme — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,20 +2,96 @@
...
+
+- (Core) `otlpreceiver`: Switch to `localhost` as the default for all endpoints. ([#8510](https://github.com/open-telemetry/opentelemetry-collector/pull/8510))
+  Disable the `component.UseLocalHostAsDefaultHost` feature gate to temporarily get the previous default.
+
+- (Splunk) `discovery`: When setting properties for discovery receiver as environment variables (`SPLUNK_DISCOVERY_*`), the values cannot reference other escaped environment variables without braces. For example, when trying to set discovery property `SPLUNK_DISCOVERY_EXTENSIONS_k8s_observer_ENABLED` to the value of another env var, `K8S_ENVIRONMENT`. For versions older than 0.104.0, setting it as `SPLUNK_DISCOVERY_EXTENSIONS_k8s_observer_ENABLED=\$K8S_ENVIRONMENT` (note the escaped variable name does not have braces) was valid. But from v0.104.0, env var names need to be passed with braces. For this example, user should modify it to `SPLUNK_DISCOVERY_EXTENSIONS_k8s_observer_ENABLED=\${K8S_ENVIRONMENT}`
+- (Contrib) `vcenterreceiver`: Drops support for vCenter 6.7 ([#33607](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33607))
```
Comment (crobert-1) [inline CHANGELOG.md:26]:
```suggestion

- (Contrib) `vcenterreceiver`: Drops support for vCenter 6.7 ([#33607](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33607))
```
Nit

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,20 +2,96 @@
...
 
 - (Splunk) Set Go garbage collection target percentage to 400% ([#5034](https://github.com/signalfx/splunk-otel-collector/pull/5034))
-  After removal of memory_ballast extension in v0.97.0, the Go garbage collection is running more aggressively, which 
-  increased CPU usage and leads to reduced throughput of the collector. This change reduces the frequency of garbage 
+  After removal of memory_ballast extension in v0.97.0, the Go garbage collection is running more aggressively, which
+  increased CPU usage and leads to reduced throughput of the collector. This change reduces the frequency of garbage
   collection cycles to improves performance of the collector for typical workloads. As a result, the collector will
   report higher memory usage, but it will be bound to the same configured limits. If you want to revert to the previous
   behavior, set the `GOGC` environment variable to `100`.
 - (Splunk) Upgrade to golang 1.21.12
```
Comment (crobert-1) [inline CHANGELOG.md:47]:
```suggestion
- (Splunk) Upgrade to golang 1.21.12 ([#5074](https://github.com/signalfx/splunk-otel-collector/pull/5074))
```

**Review by crobert-1 — APPROVED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,20 +2,106 @@
...
+
+- (Core) `otlpreceiver`: Switch to `localhost` as the default for all endpoints. ([#8510](https://github.com/open-telemetry/opentelemetry-collector/pull/8510))
+  Disable the `component.UseLocalHostAsDefaultHost` feature gate to temporarily get the previous default.
+
+- (Splunk) `discovery`: When setting properties for discovery receiver as environment variables (`SPLUNK_DISCOVERY_*`), the values cannot reference other escaped environment variables without braces. For example, when trying to set discovery property `SPLUNK_DISCOVERY_EXTENSIONS_k8s_observer_ENABLED` to the value of another env var, `K8S_ENVIRONMENT`. For versions older than 0.104.0, setting it as `SPLUNK_DISCOVERY_EXTENSIONS_k8s_observer_ENABLED=\$K8S_ENVIRONMENT` (note the escaped variable name does not have braces) was valid. But from v0.104.0, env var names need to be passed with braces. For this example, user should modify it to `SPLUNK_DISCOVERY_EXTENSIONS_k8s_observer_ENABLED=\${K8S_ENVIRONMENT}`
+<<<<<<< HEAD
+=======
+
+>>>>>>> d116609e7d13ccf9a9efe6671990f392ec433f30
```
Comment (jeffreyc-splunk) [inline CHANGELOG.md:29]:
```suggestion
```

**Review by jeffreyc-splunk — COMMENTED** [review]
Please rebase to clean up the commits in this PR.

File: CHANGELOG.md (hunk 5 above)
Comment (jinja2) [reply CHANGELOG.md]: bad merge from ide, cleaned

**Pull request merged**